engine/jobmaster/dm/api.go

```diff
@@ -55,14 +58,21 @@ func (jm *JobMaster) QueryJobStatus(ctx context.Context, tasks []string) (*JobSt
 		}
 	}
 
+	var cfgModeRevsion uint64
+	for _, task := range job.Tasks {
+		cfgModeRevsion = task.Cfg.ModRevision
+		break
+	}
+
 	var (
 		workerStatusMap = jm.workerManager.WorkerStatus()
 		wg              sync.WaitGroup
 		mu              sync.Mutex
 		jobStatus       = &JobStatus{
-			JobMasterID: jm.JobMasterID(),
-			WorkerID:    jm.ID(),
-			TaskStatus:  make(map[string]TaskStatus),
+			JobMasterID:           jm.JobMasterID(),
+			WorkerID:              jm.ID(),
+			ExpectedCfgModRevison: cfgModeRevsion,
```

Comment:
just a remind that this is conflict with the TODO "each task has its separate `cfgModeRevsion`",